debezium-connector-mysql/src/test/java/io/debezium/connector/mysql/MySqlDecimalIT.java

```diff
@@ -0,0 +1,165 @@
+                .with(MySqlConnectorConfig.TABLE_WHITELIST, DATABASE.qualifiedTableName(TABLE_NAME))
+                .with(MySqlConnectorConfig.DECIMAL_HANDLING_MODE, RelationalDatabaseConnectorConfig.DecimalHandlingMode.PRECISE)
+                .build();
+
+        start(MySqlConnector.class, config);
+
+        assertBigDecimalChangeRecord(consumeInsert());
+
+        stopConnector();
+    }
+
+    @Test
+    @FixFor("DBZ-730")
+    public void testDoubleDecimalHandlingMode() throws SQLException, InterruptedException {
+        config = DATABASE.defaultConfig()
+                .with(MySqlConnectorConfig.SNAPSHOT_MODE, MySqlConnectorConfig.SnapshotMode.INITIAL)
+                .with(MySqlConnectorConfig.TABLE_WHITELIST, DATABASE.qualifiedTableName(TABLE_NAME))
+                .with(MySqlConnectorConfig.DECIMAL_HANDLING_MODE, RelationalDatabaseConnectorConfig.DecimalHandlingMode.DOUBLE)
+                .build();
+
+        start(MySqlConnector.class, config);
+
+        assertDoubleChangeRecord(consumeInsert());
+
+        stopConnector();
+    }
+
+    @Test
+    @FixFor("DBZ-730")
+    public void testStringDecimalHandlingMode() throws SQLException, InterruptedException {
+        config = DATABASE.defaultConfig()
+                .with(MySqlConnectorConfig.SNAPSHOT_MODE, MySqlConnectorConfig.SnapshotMode.INITIAL)
+                .with(MySqlConnectorConfig.TABLE_WHITELIST, DATABASE.qualifiedTableName(TABLE_NAME))
+                .with(MySqlConnectorConfig.DECIMAL_HANDLING_MODE, RelationalDatabaseConnectorConfig.DecimalHandlingMode.STRING)
+                .build();
+
+        start(MySqlConnector.class, config);
+
+        assertStringChangeRecord(consumeInsert());
+
+        stopConnector();
+    }
+
+    private SourceRecord consumeInsert() throws InterruptedException {
+        final int numDatabase = 2;
+        final int numTables = 4;
+        final int numOthers = 1;
+
+        SourceRecords records = consumeRecordsByTopic(numDatabase + numTables + numOthers);
+
+        assertThat(records).isNotNull();
+
+        List<SourceRecord> events = records.recordsForTopic(DATABASE.topicForTable(TABLE_NAME));
+        assertThat(events).hasSize(1);
+
+        return events.get(0);
+    }
+
+    private void assertBigDecimalChangeRecord(SourceRecord record) {
+        Assertions.assertThat(record).isNotNull();
+        final Struct change = ((Struct) record.value()).getStruct("after");
+
+        Assertions.assertThat(change.get("A")).isEqualTo(new BigDecimal("1.33"));
+        Assertions.assertThat(change.get("B")).isEqualTo(new BigDecimal("-2.111"));
+        Assertions.assertThat(change.get("C")).isEqualTo(new BigDecimal("3.44400"));
+        Assertions.assertThat(change.get("D")).isEqualTo(new BigDecimal("15.28000")); // returns the default value
+        Assertions.assertThat(change.getBytes("D")).isNull(); // returns the real value, which is null
```

Comment:
these 2 lines can be confusing.  especially when getDouble and getString behave differently

Reply:
Maybe we need to fix BigDecimal handling?

Reply:
You can use `getWithoutDefault` method

Reply:
oh yes! omg. really, I just learned about that different default value behavior of get(). I would expect the opposite behaviour. get() comes without default and there is getWithDefault()... very interesting.